### ISSUE TITLE
Repetition detection

### ISSUE THREAD

**Issue by zedzedzed** [zedzedzed/table-of-contents-plus#17]
**Reported by michael on 31 Jul 2011 14:29 UTC**
Accommodate same headings by attaching a counter at the end only for two or more occurrences. Eg:

#awesome

#more_awesome

#awesome-1

#thrice_awesome

#awesome-2

One other bug: if the first heading is repeated, then the TOC will also appear in the repeated heading! One way to fix would be to use preg_replace and tell it to replace once per pattern. This would involve changing from str_replace which is much faster.

Another option is to try: http://tycoontalk.freelancer.com/php-forum/21334-str_replace-only-once-occurence-only.html

but not sure how quickly it runs.

**Comment by zedzedzed** [comment]
**Comment by michael on 2 Aug 2011 06:43 UTC**
Moving to some other milestone. This will take a while.

**Comment by zedzedzed** [comment]
**Comment by michael on 25 Aug 2011 11:40 UTC**
http://dublue.com/plugins/toc/comment-page-1/#comment-191

Suggests that duplicates on the homepage with more than one post have issues.

**Comment by zedzedzed** [comment]
**Comment by michael on 2 Nov 2011 13:16 UTC**
In [121]:
```CommitTicketReference repository="" revision="121"
re #40 addresses more NOTICES
closes #19 now supports repeated headings
```

**Comment by zedzedzed** [comment]
**Modified by michael on 2 Nov 2011 13:16 UTC**
